platforms/core-configuration/kotlin-dsl-tooling-builders/src/main/kotlin/org/gradle/kotlin/dsl/tooling/builders/KotlinBuildScriptModelBuilder.kt

```diff
@@ -94,10 +103,18 @@ data class StandardKotlinBuildScriptModel(
 
 
 internal
-object KotlinBuildScriptModelBuilder : ToolingModelBuilder {
+class KotlinBuildScriptModelBuilder(
+    private val intermediateToolingModelProvider: IntermediateToolingModelProvider
+) : ParameterizedToolingModelBuilder<KotlinBuildScriptModelParameter> {
 
     override fun canBuild(modelName: String): Boolean =
-        modelName == "org.gradle.kotlin.dsl.tooling.models.KotlinBuildScriptModel"
+        modelName == KotlinBuildScriptModel::class.qualifiedName
```

Comment:
:rocket: We were using constant strings for [a reason](https://github.com/gradle/gradle/commit/e06513c08213f57b927a3e00d5dbd75349fbffb4), at the time Kotlin reflection was expensive. I would avoid Kotlin reflection during `canBuild` methods.

Reply:
I presume a bit of Kotlin reflection should be fine at the builder creation time, right?

Also, the current `KotlinDslScriptsModelBuilder` uses the reflection in the `canBuild` call. Should I fix it as well?
https://github.com/gradle/gradle/blob/783e4aa305d675d10e2f5f56f2c5794d15356689/platforms/core-configuration/kotlin-dsl-tooling-builders/src/main/kotlin/org/gradle/kotlin/dsl/tooling/builders/KotlinDslScriptsModelBuilder.kt#L113

Reply:
Fixed

Reply:
👍